pkg/controller/job/job_controller.go

```diff
@@ -486,6 +486,12 @@ func (jm *Controller) updateJob(logger klog.Logger, old, cur interface{}) {
 	if err != nil {
 		return
 	}
+	// This is a backup operation in this case.
+	// If the job is finished, enqueue it to the orphan reconciler for finalizer removal instead of the main reconciler.
```

Comment:
```suggestion
	// The job shouldn't be marked as finished until all pod finalizers are removed, so
	// this is a backup operation.
```
Suggestion based on the original [comment](https://github.com/kubernetes/kubernetes/pull/127228/files#diff-a7577a7dede7864ff38c631319033714fdd0bed91108d976282e1099507e6ff0L482-L486).

I think the part is important to give meaning to "This is a backup operation", hinting to the checks [here](https://github.com/kubernetes/kubernetes/blob/21f54c3ca1ac87611e101f46d8b05682f51efbee/pkg/controller/job/job_controller.go#L1455-L1460). 

The last sentence "If the job is finished, enqueue it to the orphan reconciler for finalizer removal instead of the main reconciler." is rather straightforwardly following just from the code below, so I think it is safe to drop.